test/externalTests/zeppelin.sh

```diff
@@ -65,82 +65,48 @@ function zeppelin_test
 
     # Disable tests that won't pass on the ir presets due to Hardhat heuristics. Note that this also disables
     # them for other presets but that's fine - we want same code run for benchmarks to be comparable.
-    # TODO: Remove this when Hardhat adjusts heuristics for IR (https://github.com/nomiclabs/hardhat/issues/3365).
-    pushd test/utils/
-    sed -i "s|it(\('reverts \)|it.skip(\1|g" math/SafeMath.test.js
-    sed -i "s|it(\('reverts \)|it.skip(\1|g" math/SignedSafeMath.test.js
-    sed -i "s|it(\('reverts \)|it.skip(\1|g" structs/EnumerableSet.behavior.js
-    popd
-    pushd test/proxy/
-    sed -i "s|it(\('proxy admin cannot call delegated functions',\)|it.skip(\1|g" transparent/TransparentUpgradeableProxy.behaviour.js
-    popd
-
-
+    # TODO: Remove the lines below when Hardhat adjusts heuristics for IR (https://github.com/nomiclabs/hardhat/issues/3365).
```

Comment:
The issue is now closed. It was a catch-all covering all issues I reported in the past about these tests failing. They should all be passing in that case.

So it's either a problem on our side or we should open another issue about the fact that these still fail.

Can you try the repros from https://github.com/NomicFoundation/hardhat/issues/2453 and https://github.com/NomicFoundation/hardhat/issues/2115 with latest Hardhat and solc?

Also, if it's actually fixed, we can remove such workarounds from all the other external tests (this could be a separate PR though not to hold this one back).

Reply:
Related to the issue https://github.com/NomicFoundation/hardhat/issues/2453, the [first example](https://github.com/NomicFoundation/hardhat/issues/2453#issue-1159622465) is working now.
However, the [second example](https://github.com/NomicFoundation/hardhat/issues/2453#issuecomment-1059467546) fails with the same error (`Transaction reverted and Hardhat couldn't infer the reason`) for both situations that you mentioned in the comment, i.e. with and without `current()` or `increment()`.

```
Compiled 3 Solidity files successfully


  1) expectRevert
  2) expect().to.be.revertedWith
  3) [current] expectRevert
  4) [current] expect().to.be.revertedWith
  ✔ D should revert with 'ABC' (102ms)

  1 passing (950ms)
  4 failing

  1) expectRevert:

      Wrong kind of exception received
      + expected - actual

      -Transaction reverted and Hardhat couldn't infer the reason.
      +Counter: decrement overflow
      
      at expectException (node_modules/@openzeppelin/test-helpers/src/expectRevert.js:20:30)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at runNextTicks (node:internal/process/task_queues:65:3)
      at listOnTimeout (node:internal/timers:528:9)
      at processTimers (node:internal/timers:502:7)
      at expectRevert (node_modules/@openzeppelin/test-helpers/src/expectRevert.js:75:3)
      at Context.<anonymous> (test/CountersTest.js:8:3)

  2) expect().to.be.revertedWith:
     AssertionError: Expected transaction to be reverted with "Counter: decrement overflow", but other reason was found: ""
  

  3) [current] expectRevert:

      Wrong kind of exception received
      + expected - actual

      -Transaction reverted and Hardhat couldn't infer the reason.
      +Counter: decrement overflow
      
      at expectException (node_modules/@openzeppelin/test-helpers/src/expectRevert.js:20:30)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at runNextTicks (node:internal/process/task_queues:65:3)
      at listOnTimeout (node:internal/timers:528:9)
      at processTimers (node:internal/timers:502:7)
      at expectRevert (node_modules/@openzeppelin/test-helpers/src/expectRevert.js:75:3)
      at Context.<anonymous> (test/CountersTest.js:18:3)

  4) [current] expect().to.be.revertedWith:
     AssertionError: Expected transaction to be reverted with "Counter: decrement overflow", but other reason was found: ""

Your solidity settings have viaIR enabled, which is not fully supported yet. You can still use Hardhat, but some features, like stack traces, might not work correctly.
```

Probably because they do not have good support to via-ir yet: https://github.com/NomicFoundation/hardhat/issues/2453#issuecomment-1265102330

About the issue https://github.com/NomicFoundation/hardhat/issues/2115 the example is working now as well.

```
> hardhat test "test/token/ERC20/extensions/ERC20FlashMint.test.js"



  Contract: ERC20FlashMint
    flashLoan
      ✔ more than maxFlashLoan (82ms)


  1 passing (3s)
```

Reply:
So this is just something they have not fixed yet. Fine, but please update the link in the TODO:

```suggestion
    # TODO: Remove the lines below when Hardhat adjusts heuristics for IR (https://github.com/nomiclabs/hardhat/issues/3750).
```

Reply:
Thanks! I forgot about it. Update it now :)